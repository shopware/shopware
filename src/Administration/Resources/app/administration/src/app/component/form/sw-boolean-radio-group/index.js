import './sw-boolean-radio-group.scss';

const { Component } = Shopware;

/**
 * @package admin
 *
 * @private
 * @description radio input field for boolean and named entries.
 * @status ready
 * @example-type static
 * @component-example
 * <sw-boolean-radio-group
 *      label="Bool Radio group example"
 *      labelOptionTrue="Gross"
 *      labelOptionFalse="Net"
 *      :bordered="bordered">
 * </sw-boolean-radio-group>
 */
Component.register('sw-boolean-radio-group', {

    template:
`
<sw-radio-field
    class="sw-boolean-radio-group"
    v-bind="$attrs"
    :options="options"
    v-model:value="castedValue"
    name="sw-field--castedValue"
    :bordered="bordered">
</sw-radio-field>
`,

    inject: ['feature'],

    props: {
        value: {
            type: Boolean,
            required: false,
            // TODO: Boolean props should only be opt in and therefore default to false
            // eslint-disable-next-line vue/no-boolean-default
            default: true,
        },

        labelOptionTrue: {
            type: String,
            required: true,
        },

        labelOptionFalse: {
            type: String,
            required: true,
        },

        bordered: {
            type: Boolean,
            required: false,
            default: false,
        },
    },

    computed: {
        options() {
            return [
                { value: 'true', name: this.labelOptionTrue },
                { value: 'false', name: this.labelOptionFalse },
            ];
        },

        castedValue: {
            get() {
                return this.value ? this.value.toString() : 'false';
            },

            set(val) {
                this.$emit('update:value', val === 'true');
            },
        },
    },
});
