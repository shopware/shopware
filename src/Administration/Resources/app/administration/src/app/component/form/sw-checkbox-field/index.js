import template from './sw-checkbox-field.html.twig';
import './sw-checkbox-field.scss';

const { Component, Mixin } = Shopware;
const utils = Shopware.Utils;

/**
 * @public
 * @description Boolean input field based on checkbox.
 * @status ready
 * @example-type static
 * @component-example
 * <sw-checkbox-field label="Name" v-model="aBooleanProperty"></sw-checkbox-field>
 */
Component.register('sw-checkbox-field', {
    template,
    inheritAttrs: false,

    mixins: [
        Mixin.getByName('sw-form-field'),
        Mixin.getByName('remove-api-error'),
    ],

    model: {
        prop: 'value',
        event: 'change',
    },

    props: {
        disabled: {
            type: Boolean,
            required: false,
            default: false,
        },

        value: {
            type: Boolean,
            required: false,
            default: null,
        },

        inheritedValue: {
            type: Boolean,
            required: false,
            default: null,
        },

        ghostValue: {
            type: Boolean,
            required: false,
            default: null,
        },

        error: {
            type: Object,
            required: false,
            default: null,
        },

        bordered: {
            type: Boolean,
            required: false,
            default: false,
        },
    },

    data() {
        return {
            currentValue: this.value,
            id: utils.createId(),
        };
    },

    computed: {
        swCheckboxFieldClasses() {
            return {
                'has--error': this.hasError,
                'is--disabled': this.disabled,
                'is--inherited': this.isInherited,
                'sw-field__checkbox--ghost': this.ghostValue,
                'is--bordered': this.bordered,
            };
        },

        identification() {
            return `sw-field--${this.id}`;
        },

        hasError() {
            return this.error && this.error.code !== 0;
        },

        inputState() {
            return this.currentValue || false;
        },

        isInheritanceField() {
            if (this.$attrs.isInheritanceField) {
                return true;
            }
            return this.inheritedValue !== null;
        },

        isInherited() {
            if (this.$attrs.isInherited) {
                return true;
            }
            return this.isInheritanceField && this.currentValue === null;
        },
    },

    watch: {
        value() { this.currentValue = this.value; },
    },

    methods: {
        onChange(changeEvent) {
            this.$emit('change', changeEvent.target.checked);
        },
    },
});
