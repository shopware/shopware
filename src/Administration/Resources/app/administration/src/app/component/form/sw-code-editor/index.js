import Ace from 'ace-builds/src-noconflict/ace';
import { setCompleters, textCompleter } from 'ace-builds/src-noconflict/ext-language_tools';
import 'ace-builds/src-noconflict/mode-twig';
import template from './sw-code-editor.html.twig';
import './sw-code-editor.scss';

const { Component } = Shopware;
const utils = Shopware.Utils;

/**
 * @public
 * @status ready
 * @description
 * Renders a code editor
 * @example-type dynamic
 * @component-example
 * <sw-code-editor label="Description">
 * </sw-code-editor>
 */
Component.register('sw-code-editor', {
    template,

    inject: [
        'feature',
        'userInputSanitizeService',
    ],

    props: {
        value: {
            type: String,
            required: false,
            default: '',
        },

        label: {
            type: String,
            required: false,
            default: '',
        },

        completerFunction: {
            type: Function,
            required: false,
            default: null,
        },

        editorConfig: {
            type: Object,
            required: false,
            default() {
                return {};
            },
        },

        completionMode: {
            type: String,
            required: false,
            default: 'text',
            validValues: ['entity', 'text'],
            validator(value) {
                if (!value.length) {
                    return true;
                }
                return ['entity', 'text'].includes(value);
            },
        },

        mode: {
            type: String,
            required: false,
            default: 'twig',
            validValues: ['twig', 'text'],
            validator(value) {
                if (!value.length) {
                    return true;
                }
                return ['twig', 'text'].includes(value);
            },
        },

        softWraps: {
            type: Boolean,
            required: false,
            default: true,
        },

        // set focus to the component when initially mounted
        setFocus: {
            type: Boolean,
            required: false,
            default: false,
        },

        disabled: {
            type: Boolean,
            required: false,
            default: false,
        },

        sanitizeInput: {
            type: Boolean,
            required: false,
            default: false,
        },

        sanitizeFieldName: {
            type: String,
            required: false,
            default: null,
        },
    },

    data() {
        return {
            editor: {},
            editorId: utils.createId(),
            contentWasSanitized: false,
        };
    },

    computed: {
        aceConfig() {
            return {
                ...{
                    mode: `ace/mode/${this.mode}`,
                    showPrintMargin: false,
                    wrap: this.softWraps,
                    readOnly: this.disabled,
                },
                ...this.editorConfig,
            };
        },
    },

    watch: {
        value(value) {
            if (value !== null && value !== this.editor.getValue()) {
                this.editor.setValue(value, 1);
            }
        },

        softWraps() {
            this.editor.session.setOption('wrap', this.softWraps);
        },
    },

    mounted() {
        this.mountedComponent();
    },

    destroyed() {
        this.destroyedComponent();
    },

    methods: {
        mountedComponent() {
            this.editor = Ace.edit(this.$refs['editor'.concat(this.editorId)], this.aceConfig);

            this.defineAutocompletion(this.completerFunction);

            this.editor.setValue(this.value || '', 1);
            this.editor.on('input', this.onInput);
            this.editor.on('blur', this.onBlur);

            if (this.setFocus) {
                this.editor.focus();
            }
        },

        destroyedComponent() {
            delete this.editor;
        },

        onInput() {
            const value = this.editor.getValue();

            if (this.value !== value) {
                this.$emit('input', value);
            }
        },

        async onBlur() {
            const value = await this.sanitizeEditorInput(this.editor.getValue());

            this.$emit('blur', value);
        },

        async sanitizeEditorInput(value) {
            if (
                this.feature.isActive('FEATURE_NEXT_15172') &&
                this.sanitizeInput
            ) {
                this.contentWasSanitized = false;

                if (this.value !== value) {
                    try {
                        const sanitizedValue = await this.userInputSanitizeService.sanitizeInput({
                            html: value,
                            field: this.sanitizeFieldName,
                        });

                        this.contentWasSanitized = sanitizedValue?.preview !== value;

                        if (this.contentWasSanitized) {
                            this.editor.setValue(sanitizedValue?.preview ?? value, 1);
                            return this.editor.getValue();
                        }
                    } catch (ignore) { /* api endpoint did not work, keep user entry */ }
                }
            }
            return value;
        },

        defineAutocompletion(completerFunction) {
            /**
             * Sets a completer function. If completitionMode is set to "entity"
             * Autocomplete can handle [x] and . otherwise it uses the default
             * RegEx from ACE.
             * The ExecCommands sets a hook to the "insertstring" command to
             * prevent the autocompletion-popup to hide after a .
             */
            setCompleters([]);
            if (completerFunction) {
                const textCompleterClonedPlain = JSON.parse(JSON.stringify(textCompleter));
                const textCompleterCloned = JSON.parse(JSON.stringify(textCompleter));

                if (this.completionMode === 'entity') {
                    textCompleterCloned.identifierRegexps = [/[\[\]\.a-zA-Z_0-9\$\-\u00A2-\uFFFF]/];

                    textCompleterCloned.getCompletions = function getComps(editor, session, pos, prefix, callback) {
                        this.identifierRegexps = [/[\[\][a-zA-Z_0-9\$\-\u00A2-\uFFFF]/];
                        callback(null, completerFunction(prefix));
                        this.identifierRegexps = [/[\[\]\.a-zA-Z_0-9\$\-\u00A2-\uFFFF]/];
                    };

                    textCompleterCloned.completerFunction = completerFunction;
                    this.editor.completers = [textCompleterCloned];

                    const startCallback = (function startCall(e) {
                        if (e.command.name === 'insertstring') {
                            if (e.args !== '\n' && e.args !== ' ') {
                                e.editor.execCommand('startAutocomplete', null);
                            }
                        }
                    });

                    this.editor.commands.on('afterExec', startCallback);
                } else {
                    textCompleterClonedPlain.identifierRegexps = null;
                    textCompleterClonedPlain.getCompletions = function getComps(editor, session, pos, prefix, callback) {
                        callback(null, completerFunction(prefix));
                    };

                    textCompleterClonedPlain.completerFunction = completerFunction;
                    this.editor.completers = [textCompleterClonedPlain];
                }
            } else {
                this.editor.completers = [];
            }
        },
    },
});
