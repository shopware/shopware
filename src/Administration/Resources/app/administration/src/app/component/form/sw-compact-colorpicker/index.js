import template from './sw-compact-colorpicker.html.twig';
import './sw-compact-colorpicker.scss';

const { Component } = Shopware;

/**
 * @package admin
 *
 * @private
 */
Component.extend('sw-compact-colorpicker', 'sw-colorpicker', {
    template,

    emits: ['update:value'],

    inject: ['feature'],

    computed: {
        colorValue: {
            get() {
                return this.localValue;
            },
            set(newColor) {
                this.localValue = newColor;
            },
        },
    },

    methods: {
        emitColor() {
            this.$emit('update:value', this.localValue);
            this.visible = false;
        },
    },
});
