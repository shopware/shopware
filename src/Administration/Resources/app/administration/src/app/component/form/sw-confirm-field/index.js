import template from './sw-confirm-field.html.twig';
import './sw-confirm-field.scss';

const { Component } = Shopware;

/**
 * @public
 * @description Text field with additional confirmation buttons inlined in the field itself.
 * @status ready
 * @example-type static
 * @component-example
 * <sw-confirm-field placeholder="Enter value..."></sw-confirm-field>
 */
Component.register('sw-confirm-field', {
    template,

    props: {
        value: {
            type: String,
            required: false,
            default: ''
        },

        compact: {
            type: Boolean,
            required: false,
            default: false
        },

        preventEmptySubmit: {
            type: Boolean,
            required: false,
            default: false
        },

        required: {
            type: Boolean,
            required: false,
            default: false
        }
    },

    data() {
        return {
            isEditing: false,
            draft: this.value,
            event: null
        };
    },

    computed: {
        confirmFieldClasses() {
            return {
                'sw-confirm-field--compact': this.compact,
                'sw-confirm-field--editing': this.isEditing
            };
        }
    },

    watch: {
        value() {
            this.draft = this.value;
        }
    },

    methods: {
        removeActionButtons() {
            this.isEditing = false;
        },

        onStartEditing() {
            this.isEditing = true;
        },

        onBlurField({ relatedTarget }) {
            if (!!relatedTarget && relatedTarget.classList.contains('sw-confirm-field__button')) {
                return;
            }
            this.$emit('blur');
            this.cancelSubmit();
        },

        cancelSubmit() {
            this.removeActionButtons();
            this.draft = this.value;
        },

        onCancelFromKey({ target }) {
            this.cancelSubmit();
            target.blur();
        },

        onCancelSubmit() {
            this.$emit('submit-cancel');
            this.cancelSubmit();
            this.isEditing = false;
        },

        submitValue() {
            if (this.draft !== this.value) {
                this.$emit('input', this.draft, this.event);
            }
        },

        onSubmitFromKey({ target }) {
            this.event = 'key';
            this.submitValue();
            target.blur();
            this.isEditing = false;
        },

        onSubmitValue() {
            this.event = 'click';
            this.submitValue();
            this.isEditing = false;
        }
    }
});
