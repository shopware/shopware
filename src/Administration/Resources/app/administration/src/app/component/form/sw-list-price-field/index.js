import template from './sw-list-price-field.html.twig';
import './sw-list-price-field.scss';

const { Component } = Shopware;

Component.register('sw-list-price-field', {
    template,

    inheritAttrs: false,

    props: {
        price: {
            type: Array,
            required: true,
            default() {
                return [];
            },
        },

        purchasePrices: {
            type: Array,
            default() {
                return [];
            },
        },

        defaultPrice: {
            type: Object,
            required: false,
            default() {
                return {};
            },
        },

        // FIXME: add property type
        // eslint-disable-next-line vue/require-prop-types
        label: {
            required: false,
            default: true,
        },

        taxRate: {
            type: Object,
            required: true,
            default() {
                return {};
            },
        },

        currency: {
            type: Object,
            required: true,
            default() {
                return {};
            },
        },

        // FIXME: add property type
        // eslint-disable-next-line vue/require-prop-types
        compact: {
            required: false,
            default: false,
        },

        error: {
            type: Object,
            required: false,
            default: null,
        },

        // FIXME: add property type
        // eslint-disable-next-line vue/require-prop-types
        disabled: {
            required: false,
            default: false,
        },

        enableInheritance: {
            type: Boolean,
            required: false,
            default: false,
        },

        disableSuffix: {
            type: Boolean,
            required: false,
            default: false,
        },

        vertical: {
            type: Boolean,
            required: false,
            default: false,
        },

        // FIXME: add property type
        // eslint-disable-next-line vue/require-prop-types
        hideListPrices: {
            required: false,
            default: false,
        },

        showSettingPrice: {
            type: Boolean,
            required: false,
            default: true,
        },
    },

    computed: {
        priceForCurrency: {
            get() {
                const priceForCurrency = Object.values(this.price).find((price) => {
                    return price.currencyId === this.currency.id;
                });

                // check if price exists
                if (priceForCurrency) {
                    return priceForCurrency;
                }

                // otherwise calculate values
                return {
                    gross: this.convertPrice(this.defaultPrice.gross),
                    linked: this.defaultPrice.linked,
                    net: this.convertPrice(this.defaultPrice.net),
                    listPrice: this.defaultPrice.listPrice,
                };
            },
            set(newValue) {
                this.priceForCurrency.gross = newValue.gross;
                this.priceForCurrency.linked = newValue.linked;
                this.priceForCurrency.net = newValue.net;
            },
        },

        listPrice: {
            get() {
                const price = this.priceForCurrency;

                if (price.listPrice) {
                    return [price.listPrice];
                }

                return [{
                    gross: null,
                    currencyId: this.defaultPrice.currencyId ? this.defaultPrice.currencyId : this.currency.id,
                    linked: true,
                    net: null,
                }];
            },

            set(newValue) {
                const price = this.priceForCurrency;

                if (price) {
                    this.$set(price, 'listPrice', newValue);
                }
            },
        },

        defaultListPrice() {
            const price = this.defaultPrice.listPrice;

            if (price) {
                return price;
            }

            return {
                currencyId: this.defaultPrice.currencyId ? this.defaultPrice.currencyId : this.currency.id,
                gross: null,
                net: null,
                linked: true,
            };
        },

        isInherited() {
            const priceForCurrency = Object.values(this.price).find((price) => {
                return price.currencyId === this.currency.id;
            });

            return !priceForCurrency;
        },

        listPriceHelpText() {
            if (!this.vertical || this.compact) {
                return null;
            }

            return this.$tc('global.sw-list-price-field.helpTextListPriceGross');
        },
    },

    methods: {
        listPriceChanged(value) {
            if (Number.isNaN(value.gross) || Number.isNaN(value.net)) {
                value = null;
            }

            this.listPrice = value;
        },

        convertPrice(value) {
            const calculatedPrice = value * this.currency.factor;
            const priceRounded = calculatedPrice.toFixed(this.currency.decimalPrecision);
            return Number(priceRounded);
        },
    },
});
