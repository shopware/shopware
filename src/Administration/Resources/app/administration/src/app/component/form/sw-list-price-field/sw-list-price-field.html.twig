<!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
{% block sw_list_price_field %}
<div
    :class="{
        'sw-list-price-field': true,
        'sw-list-price-field__vertical': vertical
    }"
>

    <div class="sw-list-price-field__price">
        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
        {% block sw_list_price_field_price %}
        <sw-price-field
            class="sw-list-price-field__price-field"
            :price="price"
            required
            :compact="compact"
            :disable-suffix="disableSuffix"
            :default-price="defaultPrice"
            :tax-rate="taxRate"
            :error="error"
            :disabled="disabled"
            :enable-inheritance="enableInheritance"
            :label="label"
            :currency="currency"
        />
        {% endblock %}
    </div>

    <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
    {% block sw_product_price_form_purchase_price_field %}
    <div
        v-if="!hidePurchasePrices && showSettingPrice"
        class="sw-list-price-field__purchase-price"
    >
        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
        {% block sw_product_price_form_purchase_price_field_content %}
        <sw-price-field
            class="sw-purchase-price-field"
            :price="purchasePrices"
            :gross-label="$tc('sw-product.priceForm.labelPurchasePriceGross')"
            :net-label="$tc('sw-product.priceForm.labelPurchasePriceNet')"
            :tax-rate="taxRate"
            :disabled="disabled"
            :currency="currency"
        />
        {% endblock %}
    </div>
    {% endblock %}

    <div class="sw-list-price-field__list-price">
        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
        {% block sw_list_price_field_list_price %}
        <sw-price-field
            v-if="!hideListPrices && showSettingPrice"
            :price="listPrice"
            :default-price="defaultListPrice"
            :tax-rate="taxRate"
            :disabled="disabled"
            allow-empty
            :gross-label="$tc('global.sw-list-price-field.labelListPriceGross')"
            :gross-help-text="listPriceHelpText"
            :net-label="$tc('global.sw-list-price-field.labelListPriceNet')"
            :compact="compact"
            :disable-suffix="disableSuffix"
            :label="label"
            :enable-inheritance="enableInheritance"
            :inherited="isInherited"
            class="sw-list-price-field__list-price-field"
            :currency="currency"
            @change="listPriceChanged"
        />
        {% endblock %}
    </div>

    <div class="sw-list-price-field__regulation-price">
        <!-- eslint-disable-next-line sw-deprecation-rules/no-twigjs-blocks -->
        {% block sw_list_price_field_regulation_price %}
        <sw-price-field
            v-if="!hideRegulationPrices && showSettingPrice"
            :price="regulationPrice"
            :default-price="defaultRegulationPrice"
            :tax-rate="taxRate"
            :disabled="disabled"
            allow-empty
            :gross-label="$tc('global.sw-list-price-field.labelRegulationPriceGross')"
            :gross-help-text="regulationPriceHelpText"
            :net-label="$tc('global.sw-list-price-field.labelRegulationPriceNet')"
            :compact="compact"
            :disable-suffix="disableSuffix"
            :label="label"
            :enable-inheritance="enableInheritance"
            :inherited="isInherited"
            class="sw-list-price-field__regulation-price-field"
            :currency="currency"
            @change="regulationPriceChanged"
        />
        {% endblock %}
    </div>
</div>
{% endblock %}
