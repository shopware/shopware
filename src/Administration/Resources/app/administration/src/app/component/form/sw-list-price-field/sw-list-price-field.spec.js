/**
 * @package admin
 */

import { shallowMount } from '@vue/test-utils';
import 'src/app/component/form/sw-list-price-field';

// mock data
const purchasePrices = {
    currencyId: 'a435755c6c4f4fb4b81ec32b4c07e06e',
    net: 20,
    gross: 25,
    linked: false
};
const dollarPrice = {
    currencyId: 'a435755c6c4f4fb4b81ec32b4c07e06e',
    net: 250,
    gross: 123,
    linked: false
};
const euroPrice = {
    currencyId: 'b7d2554b0ce847cd82f3ac9bd1c0dfca',
    net: 152.33644859813083,
    gross: 163,
    linked: true
};

const taxRate = {
    name: '7%',
    taxRate: 7,
    id: 'd9eac12a83984df59a618a5be1342009'
};

const currency = {
    id: 'a435755c6c4f4fb4b81ec32b4c07e06e',
    name: 'US-Dollar',
    isoCode: 'USD',
    decimalPrecision: 2,
    factor: 1.17085,
    shortName: 'USD',
    symbol: '$'
};

const defaultPrice = {
    currencyId: 'b7d2554b0ce847cd82f3ac9bd1c0dfca',
    gross: 163,
    net: 152.33644859813083,
    linked: true
};

// initial component setup
const setup = async (propOverride) => {
    const propsData = {
        price: [dollarPrice, euroPrice],
        purchasePrices: [purchasePrices],
        taxRate,
        currency,
        defaultPrice,
        enableInheritance: false,
        ...propOverride
    };

    return shallowMount(await Shopware.Component.build('sw-list-price-field'), {
        stubs: ['sw-price-field'],
        propsData
    });
};

describe('components/form/sw-list-price-field', () => {
    it('should be a Vue.js component', async () => {
        const wrapper = await setup();
        expect(wrapper.vm).toBeTruthy();
    });

    it('should be rendered correctly', async () => {
        const wrapper = await setup();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should set listPrice null when the gross value is NaN', async () => {
        const wrapper = await setup();
        const listPrice = {
            currencyId: 'b7d2554b0ce847cd82f3ac9bd1c0dfca',
            gross: parseFloat(''),
            linked: true,
            net: 1
        };
        await wrapper.vm.listPriceChanged(listPrice);
        expect(wrapper.vm.priceForCurrency.listPrice).toBeNull();
    });

    it('should set listPrice null when the net value is NaN', async () => {
        const wrapper = await setup();
        const listPrice = {
            currencyId: 'b7d2554b0ce847cd82f3ac9bd1c0dfca',
            gross: 1,
            linked: true,
            net: parseFloat('')
        };
        await wrapper.vm.listPriceChanged(listPrice);
        expect(wrapper.vm.priceForCurrency.listPrice).toBeNull();
    });

    it('should set purchasePrice to default value when the input purchasePrices is empty', async () => {
        const wrapper = await setup({ hidePurchasePrices: true });
        await wrapper.vm.$nextTick();

        expect(wrapper.find('.sw-list-price-field__purchase-price').exists()).toBeFalsy();
    });

    it('should set the correct inherited state when inherited', async () => {
        const wrapper = await setup();
        await wrapper.setProps({
            price: [euroPrice]
        });

        expect(wrapper.vm.isInherited).toBeTruthy();
    });

    it('should set the correct inherited state when not inherited', async () => {
        const wrapper = await setup();

        await wrapper.setProps({
            price: [dollarPrice]
        });

        expect(wrapper.vm.isInherited).toBeFalsy();
    });

    it('should not display gross help text when not in vertical mode', async () => {
        const wrapper = await setup();

        expect(wrapper.find('.sw-list-price-field__list-price sw-price-field-stub')
            .attributes()['gross-help-text']).toBeUndefined();
    });

    it('should display gross help text when in vertical mode', async () => {
        const wrapper = await setup({
            vertical: true
        });

        expect(wrapper.find('.sw-list-price-field__list-price sw-price-field-stub')
            .attributes()['gross-help-text']).toBe('global.sw-list-price-field.helpTextListPriceGross');
    });

    it('should not display gross help text when in compact mode', async () => {
        const wrapper = await setup({
            vertical: true,
            compact: true
        });

        expect(wrapper.find('.sw-list-price-field__list-price sw-price-field-stub')
            .attributes()['gross-help-text']).toBeUndefined();

        expect(wrapper.find('.sw-list-price-field__regulation-price sw-price-field-stub')
            .attributes()['gross-help-text']).toBeUndefined();
    });
});
