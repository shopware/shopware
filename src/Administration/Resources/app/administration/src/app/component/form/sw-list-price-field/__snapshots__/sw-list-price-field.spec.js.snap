// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/sw-list-price-field should be rendered correctly 1`] = `
<div
  class="sw-list-price-field"
>
  <div
    class="sw-list-price-field__price"
  >
    <sw-price-field-stub
      class="sw-list-price-field__price-field"
      currency="[object Object]"
      default-price="[object Object]"
      label="true"
      required=""
      tax-rate="[object Object]"
      value="[object Object],[object Object]"
    />
  </div>
   
  <div
    class="sw-list-price-field__purchase-price"
  >
    <sw-price-field-stub
      class="sw-purchase-price-field"
      currency="[object Object]"
      gross-label="sw-product.priceForm.labelPurchasePriceGross"
      net-label="sw-product.priceForm.labelPurchasePriceNet"
      tax-rate="[object Object]"
      value="[object Object]"
    />
  </div>
   
  <div
    class="sw-list-price-field__list-price"
  >
    <sw-price-field-stub
      allow-empty=""
      class="sw-list-price-field__list-price-field"
      currency="[object Object]"
      default-price="[object Object]"
      gross-label="global.sw-list-price-field.labelListPriceGross"
      label="true"
      net-label="global.sw-list-price-field.labelListPriceNet"
      tax-rate="[object Object]"
      value="[object Object]"
    />
  </div>
   
  <div
    class="sw-list-price-field__regulation-price"
  >
    <sw-price-field-stub
      allow-empty=""
      class="sw-list-price-field__regulation-price-field"
      currency="[object Object]"
      default-price="[object Object]"
      gross-label="global.sw-list-price-field.labelRegulationPriceGross"
      label="true"
      net-label="global.sw-list-price-field.labelRegulationPriceNet"
      tax-rate="[object Object]"
      value="[object Object]"
    />
  </div>
</div>
`;
